File: CURRENT/c-sharp/tests-and-release/5-Working-Tests/WorkingTests.cs
using System;
using org.mariuszgromada.math.mxparser;

class WorkingTests
{
	static void Main(string[] args)
	{
		Argument x = new Argument("x = 10");
		Expression e = new Expression("1+2*x", x);
		mXparser.consolePrintln(e.getExpressionString() + " = " + e.calculate());
	}
}

File: CURRENT/cpp/working-tests/WorkingTests.cpp
#include "org/mariuszgromada/math/mxparser.hpp"

int main() {
	ConstantPtr c = new_Constant("c = 2");
	ArgumentPtr a = new_Argument("a = 2*c", c);
	ArgumentPtr b = new_Argument("b = a+c", a, c);
	FunctionPtr f = new_Function("f(x,y)= x*y");
	ExpressionPtr e = new_Expression("2+sin(b)-f(a,c)", a, b, c, f);
	mXparser::consolePrintln(e->getExpressionString() + " = " + e->calculate());
	mXparser::consolePrintln(e->getErrorMessage());
}